### ISSUE TITLE
Update for TWW?

### ISSUE THREAD

**Issue by Taritara** [Vladinator89/wow-addon-miniloot#6]
I love your MiniLoot addon, is it possible to get it updated for TWW? Currently, it is not working for 11.0

**Comment by ynazar1** [comment]
In options.lua:924 or so replace `InterfaceOptions_AddCategory(loaded)` with:
```
		if InterfaceOptions_AddCategory then
			InterfaceOptions_AddCategory(loaded)
		else
			local category, layout = Settings.RegisterCanvasLayoutCategory(loaded, loaded.name);
			Settings.RegisterAddOnCategory(category);
			settingsCategory = category
		end
```

**Comment by Taritara** [comment]
Thank you very much!

**Comment by ynazar1** [comment]
Add on the top of every file that errors in 11.0.2
```
local IsAddOnLoaded = C_AddOns.IsAddOnLoaded
local GetAddOnMetadata = C_AddOns.GetAddOnMetadata
```

This mostly resolves errors with api changes. Some other stuff with items and whatnot but I don't believe that there are any other stuff in miniloot. 

I had to fix like 20 addons just now so that my UI isn't flooded with errors.

**Comment by Taritara** [comment]
Many thanks!

**Comment by ynazar1** [comment]
@Vladinator should be able to merge changes above, bump TOC and release it. It's fully functional best i can tell.

**Comment by Vladinator** [comment]
Sorry that I didn't respond earlier.

I have been working on a rewrite, as the original addon was made a very long time ago, maintaining the old code was getting rough.

I have not yet pushed the latest rewrite, as it is missing some settings UI to control the custom filter rules.

I made a `v1` branch that contains the original addon with fixes so it works with TWW as-is.

The `master` branch is now containing the new code that I could release as a beta release of the addon.

TLDR; The update today should contain the fixes so it works with TWW without those errors.

**Comment by ynazar1** [comment]
Awesome and thank you. Fingers crossed, the minor fixes will tie us over for a bit and let you work without worrying if it's still functional or not in early TWW. 

Sent from my mobile headquarters.  
 
  On Sun, Aug 18, 2024 at 2:16 AM, Alex ***@***.***> wrote:   


Sorry that I didn't respond earlier.

I have been working on a rewrite, as the original addon was made a very long time ago, maintaining the old code was getting rough.

I have not yet pushed the latest rewrite, as it is missing some settings UI to control the custom filter rules.

I made a v1 branch that contains the original addon with fixes so it works with TWW as-is.

The master branch is now containing the new code that I could release as a beta release of the addon.

TLDR; The update today should contain the fixes so it works with TWW without those errors.
